packages/react-router/src/ReactRouter/IonRouter.tsx

```diff
@@ -187,7 +187,7 @@ class IonRouterInner extends React.PureComponent<IonRouteProps, IonRouteState> {
           routeInfo.lastPathname = currentRouteInfo?.pathname || routeInfo.lastPathname;
           routeInfo.prevRouteLastPathname = currentRouteInfo?.lastPathname;
           routeInfo.pushedByRoute = pushedByRoute;
-          routeInfo.routeDirection = currentRouteInfo?.routeDirection || routeInfo.routeDirection;
+          routeInfo.routeDirection = routeInfo.routeDirection || currentRouteInfo?.routeDirection;
           routeInfo.routeAnimation = currentRouteInfo?.routeAnimation || routeInfo.routeAnimation;
```

Comment:
Note: May want to do the same thing for routeAnimation, but I was trying to be surgical because I haven't tested a change to `routeAnimation` and I am not an Ionic routing expert :)